securedrop/journalist.py

```diff
@@ -581,16 +581,17 @@ def bulk():
 
     doc_names_selected = request.form.getlist('doc_names_selected')
     selected_docs = [doc for doc in g.source.collection
-                     if doc.filename in doc_names_selected]
-
+                     if (doc.filename in doc_names_selected) or action == 'download_all']
```

Comment:
I'd prefer it if you didn't bury the `download_all` logic in the list comprehension. I think this might be better:

```py
if action == 'download_all':
    selected_docs = g.source.collection
else:
    # list comprehension to filter based on doc_names_selected
```